src/shipit_uplift/shipit_uplift/api.py

```diff
@@ -376,18 +381,25 @@ def coverage_by_dir(path=''):
 
 
 def coverage_by_changeset(changeset):
-    try:
-        return coverage_by_changeset_impl.generate(changeset)
-    except Exception as e:
+    job = q.fetch_job(changeset)
+
+    if job is None:
+        job = q.enqueue(coverage_by_changeset_impl.generate, changeset, job_id=changeset, result_ttl=86400)
+
+    if job.result is not None:
```

Comment:
`is not None` is redundant. Same below.

Reply:
I'm using `is not None` because the property is set to `None` when there is no result and to a value when there is a result.
if there's a bug somewhere and the result is e.g. `{}`, we will return it. If we didn't have `is not None` we would return 202 and the client would keep requesting indefinitely.

Reply:
This, right?
Thanks for explaining!
```python
In [5]: for i in (None, {}):
   ...:     if i:
   ...:         print("We're in")
   ...:     if i is not None:
   ...:         print("Yahoo!")
   ...:
Yahoo!
```